### PR TITLE
Add welcome mat with homepage redirects

### LINKED ISSUE #529: Create "welcome mat" for authenticated users
## Acceptance criteria

- [x] Cards: USWDS card component. Button links to users and billing, respectively. If they don't have access to one, don't show the card. 
  - If I don't have permission to card (users or billing), adjust other cards (expand or move over - judgement call). 
- [x] Users card: live update for number of users in currently picked organization (see #410).
  -  If I'm on org A and choose Org B from the picker, update the Dashboard with Org B info.
- [x] Blog: automatically pull in the latest entry from https://cloud.gov/news/ via feed
  - currently moving to 11ty 
- [ ] Footer: Updated version ❗ 
- [x] Mobile: adjusted sizing/spacing 

⚠️ Remember, there is no "logged out" welcome mat view.  

 _Originally posted by @beepdotgov in [#477](https://github.com/cloud-gov/cg-ui/issues/477#issuecomment-2353386573)_

### PR DESCRIPTION
**Pull request by echappen**
⚠️ **Add environment variable to [cg-ui](https://cg-ui.dev.us-gov-west-1.aws-us-gov.cloud.gov/) before merging/deploying:** 
```
NEXT_PUBLIC_BLOG_FEED_URL=https://cloud.gov/updates.xml
```

## Changes proposed in this pull request:

- Redirects users to `/login` when navigating to root `/` and not authenticated
- When authenticated and navigating to root, redirect users to one of their orgs: `/orgs/:org_id`
  - When `lastViewedOrgId` is set, take them to this org
  - If not, then make a CAPI request to get their first org and take them there
- Implements welcome mat design for `/orgs/:org_id` as specified in #529 
- Pulls in real blog data from `https://cloud.gov/updates.xml`

### How to test
- add `NEXT_PUBLIC_BLOG_FEED_URL=https://cloud.gov/updates.xml` to `.env.local`
- Spin up a `uaa-docker` container ([instructions](https://github.com/cloud-gov/cg-ui/blob/main/uaa-docker/README.md))
- Log into the CF CLI
- Run the dev server `npm run dev-cf`
- Try to access `/` - you should be redirected to UAA login
- Use the credentials found in `/uaa-docker/uaa.yml` to log in
- After log in, you should be redirected to `/orgs/:org_id`

### Related issues

Closes #529 

### Submitter checklist

- [ ] Added logging is not capturing sensitive data and is set to an appropriate level (DEBUG vs INFO etc)
- [ ] Updated relevant documentation (README, ADRs, explainers, diagrams)

## Security considerations

Uses org id stored in cookie to make CF API requests in order to show the content on the page

### CODE REVIEW

**Comment by beepdotgov** [comment]
@echappen This looks _so good_ already. Excited to see this (almost) live!

I spotted a couple things in this branch; I know it’s still a WIP, so logging them here for when/if you’re ready for them.

This is probably a setup thing on my end, but the blog area’s not loading for me:

image

The leading on the main card is a little tight; could we increase it a bit?

Screenshot 2024-11-05 at 11 30 29 AM

(Adding a `line-height-sans-4` to that graf looked good to me, but let me know what you think.)

It looks like the org picker is overlapping the main content area:

Screenshot 2024-11-05 at 11 26 53 AM

And finally, will you be working the updated footer into this PR? ([widescreen](https://www.figma.com/design/DNbFdd7YrA2MpgeXNgmNLz/Cloud.gov-%E2%80%94-Dashboard-UI-redesign?node-id=1818-13386&t=8rYS2I6PsVcjpple-4), [small screen](https://www.figma.com/design/DNbFdd7YrA2MpgeXNgmNLz/Cloud.gov-%E2%80%94-Dashboard-UI-redesign?node-id=1818-13475&t=8rYS2I6PsVcjpple-4))

(If it’s out of scope for this week, that’s _great_ — let me know, and I’ll open a separate issue for implementation!)

**Comment by echappen** [comment]
@beepdotgov Thanks for all this! Let's open up a separate issue to update the footer, as I think that will take some work.

**Comment by beepdotgov** [comment]
@echappen On it!

**Comment by beepdotgov** [comment]
Thanks so much for the tip on the blog feed config, @echappen — it looks _great_ now.

I’d love to suggest a couple tiny tweaks, if you have a minute! Here’s the current state:

current

Here’s a quick in-browser edit:

tweaks

Here’s the TLDR of what I changed:

1. Changed `.text-light` to `.text-normal` on the h2
2. Added a non-breaking space between “Cloud.gov” and “blog” in the h2, to prevent the orphaned word
3. Removed the top margin on the article title
4. Added `line-height-sans-4` to the blog teaser

Do those seem doable to you?

(Also, it looks like the browser’s default link colors might be getting applied to links in that block; is that right? Can we apply our default blue for the dash?)

**Comment by echappen** [comment]
@beepdotgov All doable! Just pushed those updates.

File: src/app/orgs/[orgId]/page.tsx [hunk 1]
```diff
@@ -1,46 +1,66 @@
-/*
-  These route segment configs should bust Data and Route caching
-  So that the list of users is always up-to-date.
-  More info: https://nextjs.org/docs/app/building-your-application/caching#segment-config-options
-*/
-export const dynamic = 'force-dynamic';
-export const revalidate = 0;
+import Image from 'next/image';
```
Comment (hursey013) [inline src/app/orgs/[orgId]/page.tsx:1]: We need to use our custom `Image` component in order to remove the `style="color:transparent"` the Next.js component adds to image.

File: src/components/BlogSnippet/BlogSnippet.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { getBlogFeed, BlogObj } from '@/api/blog/blog';
+import { formatDate } from '@/helpers/dates';
+import Link from 'next/link';
+import Image from 'next/image';
```
Comment (hursey013) [inline src/components/BlogSnippet/BlogSnippet.tsx:4]: We need to use our custom `Image` component in order to remove the `style="color:transparent"` the Next.js component adds to image.

File: src/components/PreFooter.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import Link from 'next/link';
+import Image from 'next/image';
```
Comment (hursey013) [inline src/components/PreFooter.tsx:2]: We need to use our custom `Image` component in order to remove the `style="color:transparent"` the Next.js component adds to image.

**Comment by echappen** [comment]
> I'm noticing on initial login that I'm directed to a blank homepage as opposed to the login page. If I reload the page, I'm properly directed to localhost:9001/login

@hursey013 I noticed that middleware wasn't triggering at all for initial page loads of the root url `/`. Including it in the route matchers seems to fix that. Do you know if we'd need anything more specific for [the matcher](https://github.com/cloud-gov/cg-ui/pull/583/commits/6ea202c1b393b0f660fe928ced1c316b0d909c17)?

**Comment by hursey013** [comment]
> > I'm noticing on initial login that I'm directed to a blank homepage as opposed to the login page. If I reload the page, I'm properly directed to localhost:9001/login
> 
> @hursey013 I noticed that middleware wasn't triggering at all for initial page loads of the root url `/`. Including it in the route matchers seems to fix that. Do you know if we'd need anything more specific for [the matcher](https://github.com/cloud-gov/cg-ui/pull/583/commits/6ea202c1b393b0f660fe928ced1c316b0d909c17)?

It seems like `/` should get matched with the regex based on the negative (lookahead option)[https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher] but it doesn't seem like there would be any harm in you adding it explicitly like you have and it seems to be a solution offered here: https://github.com/vercel/next.js/issues/62078

**Comment by echappen** [comment]
@hursey013 Oh, nice find!

**Review by hursey013 — APPROVED** [review]
This looks great, thank you Eleni! Only thought would be whether we need to include `withLandingPage`, on one hand I like the extra peace of mind, but I could also see that creating some confusion in the future, I'll defer to you on what you think is best!

**Pull request merged**